File: kandoe_service_session/src/Models/Models/Imported/Theme.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson.Serialization.Attributes;

namespace Models.Models.Imported
{
    public class Theme : ITheme
    {
        [BsonId]
        public string ThemeId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public ICollection<Card> Cards { get; set; }
        public ICollection<string> Organizers {get; set;}
        public DateTime UpdatedOn { get; set; } = DateTime.Now;
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public bool IsEnabled { get; set; } = true;
        public bool Public { get; set; }
        public ICollection<string> Tags { get; set; }
        public string Username { get; set; }

        public Theme()
        {
            ThemeId = Guid.NewGuid().ToString();
            UpdatedOn = DateTime.UtcNow;
            CreatedOn = DateTime.UtcNow;
            IsEnabled = true;
        }
    }
}

File: kandoe_angular2/src/app/domain/theme.ts
import {Card} from "./card";

export interface Theme {
  themeId: string;
  name: string;
  description: string;
  cards?: Card[];
  organizers: string[];
  updatedOn?: Date;
  createdOn?: Date;
  isEnabled?: boolean;
  isPublic: boolean;
  tags?: string[];
  username: string;
}
